include/date_time.h

```diff
@@ -91,6 +91,20 @@ int date_time_uptime_to_unix_time_ms(int64_t *uptime);
  */
 int date_time_now(int64_t *unix_time_ms);
 
+/** @brief Get the current date time in local time.
+ *
+ *  @note If the function fails, the passed in variable retains its
+ *        old value.
+ *
+ *  @param[out] local_time_ms Pointer to a variable to store the current date
+ *                            time.
+ *
+ *  @return 0        If the operation was successful.
+ *  @return -ENODATA If the library does not have a valid date time.
+ *  @return -EINVAL  If the passed in pointer is NULL.
+ */
+int date_time_now_local(int64_t *local_time_ms);
```

Comment:
If the timezone is not available within the library when modem doesn't give it, or NTP is used, this function will basically return UTC time (0 time zone). I'm thinking if that's what we want without knowing about not really necessarily being at UTC.
We could return some error code and still fill the `local_time_ms` with the UTC time and application could determine what to do.
This is just a thought and feel free to comment? Also interested in comments from @tokangas.

Reply:
I like the idea of returning a specific error in case the timezone is not known. I agree, that the user of the library should somehow be made aware of this.